Wire up ChIP ingress client for TelemetryIngress

pkcll opened the pull request:
## What

- Add chipIngressClient field to telemetry Manager struct
- Pass chipingress.Client to NewManager constructor
- Add ChipIngressEnabled option to TelemetryIngress config

Related PR's 
- https://github.com/smartcontractkit/chainlink-common/pull/1647
  - https://github.com/smartcontractkit/chainlink-common/pull/1646

## Why 

This is prep work for OTI Telemetry migration to ChIP ingress

pkcll commented on core/config/toml/types.go:
```diff
@@ -730,14 +730,15 @@ func (d *DatabaseBackup) setFrom(f *DatabaseBackup) {
+	SendTimeout        *commonconfig.Duration
+	UseBatchSend       *bool
+	Endpoints          []TelemetryIngressEndpoint `toml:",omitempty"`
+	ChipIngressEnabled *bool
```
Flag to enable OTI telemetry ingress via ChIP Ingress

pkcll commented on core/services/telemetry/manager.go:
```diff
@@ -32,6 +33,8 @@ type Manager struct {
 	useBatchSend                bool
 	MonitoringEndpointGenerator MonitoringEndpointGenerator
+
+	chipIngressClient chipingress.Client
```
Add reference to chip ingress client to be used for OTI telemetry ingress

pkcll commented on core/services/chainlink/application.go:
```diff
@@ -440,7 +441,12 @@ func NewApplication(ctx context.Context, opts ApplicationOpts) (Application, err
+		globalLogger.Info("ChIP Ingress is enabled for telemetry")
+		chipIngressClient = beholder.GetClient().Chip
+	}
+	telemetryManager := telemetry.NewManager(cfg.TelemetryIngress(), csaKeystore, globalLogger, chipIngressClient)
```
Wire up chip ingress client for OTI telemetry ingress Manager

jmank88 commented on core/services/chainlink/application.go:
```diff
@@ -440,7 +441,12 @@ func NewApplication(ctx context.Context, opts ApplicationOpts) (Application, err
+	if cfg.TelemetryIngress().ChipIngressEnabled() {
+		globalLogger.Info("ChIP Ingress is enabled for telemetry")
+		chipIngressClient = beholder.GetClient().Chip
+	}
```
Can we do this logic inside of `telemetry.NewManger`? We already pass the config, and that is the only place we pass the client.

pkcll replied:
Addressed https://github.com/smartcontractkit/chainlink/pull/20063/commits/0bfcbf6fa23360503cf04cb15e654dcecaa97c38

jmank88 commented on core/services/chainlink/config_telemetry_ingress.go:
```diff
@@ -56,6 +56,13 @@ func (t *telemetryIngressConfig) Endpoints() []config.TelemetryIngressEndpoint {
+func (t *telemetryIngressConfig) ChipIngressEnabled() bool {
+	if t.c.ChipIngressEnabled == nil {
+		return false
+	}
```
Please don't hard-code defaults. The TOML config definitions should be used.
```suggestion
```

pkcll replied:
https://github.com/smartcontractkit/chainlink/pull/20063/commits/0cc74c0181911a89f4fdb0ff744a5f2c3d6fc84e

pkcll committed:
feat: Wire up ChIP ingress client for TelemetryIngress

- Add chipIngressClient field to telemetry Manager struct
- Pass chipingress.Client to NewManager constructor
- Add ChipIngressEnabled option to TelemetryIngress config

pkcll committed:
fix test
